includes/Pages/PageListFlaggedComments.php

```diff
@@ -36,15 +36,33 @@ protected function main()
 
         $seeRestrictedComments = $this->barrierTest('seeRestrictedComments', $currentUser, 'RequestData');
         $seeCheckuserComments = $this->barrierTest('seeCheckuserComments', $currentUser, 'RequestData');
+        $alwaysSeePrivateData = $this->barrierTest('alwaysSeePrivateData', $currentUser, 'RequestData');
 
         foreach ($commentObjects as $object) {
             $data = [
                 'id'            => $object->getId(),
                 'visibility'    => $object->getVisibility(),
                 'updateversion' => $object->getUpdateVersion(),
-                'hidden'        => false
+                'hidden'        => false,
+                'unreserved'    => false,
             ];
 
+
+            if (!$alwaysSeePrivateData) {
```

Comment:
Perhaps I'm not understanding your point here.

The `RequestData::alwaysSeePrivateData` permission already exists, and is already in widespread use in Production today.

For wider context, private data is always hidden on the tool unless (at least) one of three situations is true:
* The user has reserved the request the private data refers to
* The user has browsed to the hash link for the request, having received this from whoever reserved the request
* The user has the `RequestData::alwaysSeePrivateData` permission via one of the roles granted that permission - currently `toolRoot`, `admin`, or `checkuser`

Are you suggesting that this third point should be removed entirely (noting that it would remove the exemption from CUs and admins and force members of those groups to reserve each request before seeing the data)?

This patch more widely hides the content of flagged comments from those who are not able to see private data, per your request in https://github.com/enwikipedia-acc/waca/issues/777#issuecomment-1360758330. This effectively treats flagged comments as private data.

Historically, ListFlaggedComments has had no private data on it by design, so it's not needed any of the private data protection that pages like ViewRequest have required. Thus at the moment (before this patch), the `ListFlaggedComments` page bypasses this protection entirely.

While we could make the (currently correct) assumption that everyone with the ability to browse to that page (by virtue of having the permission `PageListFlaggedComments::main`) also holds `RequestData::alwaysSeePrivateData`, and thus continue to skip the checks, I really don't want to rely on assumptions and thus I'm explicitly calling out and have written code for the case where that assumption isn't correct.

I think that not granting access to this page alongside alwaysSeePrivateData is a stupid configuration (hence my comment) as it forces the page to degrade to effectively a list of requests which happen to have flagged comments, rather than listing the comments in-line for quicker review.